saw-core/src/Verifier/SAW/SharedTerm.hs

```diff
@@ -2622,6 +2623,31 @@ scUnfoldConstantSet sc b names t0 = do
           _ -> scTermF sc =<< traverse go tf
   go t0
 
+scUnfoldOnceFixConstantSet :: SharedContext
+                           -> Bool  -- ^ True: unfold constants in set. False: unfold constants NOT in set
+                           -> Set VarIndex -- ^ Set of constant names
+                           -> Term
+                           -> IO Term
+scUnfoldOnceFixConstantSet sc b names t0 = do
+  cache <- newCache
+  let go :: Term -> IO Term
+      go t@(Unshared tf) =
+        case tf of
```

Comment:
Beginning with this `case` statement, both this and the following `go` case look identical except the re-wrapping of the catch-all case.  There's enough complexity here that it might be nice to extract these bodies to a single helper function in this same `let` binding context.